src/MatLang.jl

```diff
@@ -1,23 +1,215 @@
+################################################################
+################################################################
+"""
+    zerosM(idx)         # square idx*idx matrix
+    zerosM(Type,idx)    # square idx*idx matrix
+    zerosM(sizeAsArray) # non-efficient Matlab way
+    zerosM(Type, sizeAsArray) # non-efficient Matlab way
+
+# Example
+mZeros0=zerosM(2); # same as zeros(2,2)
+
+mZeros1=zerosM(2,2); # same as zeros(2,2)
+
+mZeros2=zerosM(Int32,2,2); # same as zeros(Int32,2,2)
+
+mZeros3=zerosM((2,2)); # giving size as Tuple
+
+mZeros4=zerosM(Int32,(2,2)); # giving size as Tuple and also the Int32 type
+
+mZeros5=zerosM([2,2]); # giving size as an Array, non-efficient Matlab way. Array should be Integer otherwise you will get errors.
+
+mZeros6=zerosM(Int32,[2,2]); # giving size as Array, non-efficient Matlab way
+"""
+zerosM(dim::Integer)=zeros(dim,dim)
+zerosM(a::Array)=zeros(Tuple(a))
+zerosM(T::Type,a::Array)=zeros(T::Type,Tuple(a))
+zerosM(args...) = zeros(args...)
+################################################################
+"""
+    onesM(idx)         # square idx*idx matrix
+    onesM(Type,idx)    # square idx*idx matrix
+    onesM(sizeAsArray) # non-efficient Matlab way
+    onesM(Type, sizeAsArray) # non-efficient Matlab way
+
+# Example
+mOnes0=onesM(2); # same as ones(2,2)
+
+mOnes1=onesM(2,2); # same as ones(2,2)
+
+mOnes2=onesM(Int32,2,2); # same as ones(Int32,2,2)
+
+mOnes3=onesM((2,2)); # giving size as Tuple
+
+mOnes4=onesM(Int32,(2,2)); # giving size as Tuple and also the Int32 type
+
+mOnes5=onesM([2,2]); # giving size as Array, non-efficient Matlab way. Array should be Integer otherwise you will get errors.
+
+mOnes6=onesM(Int32,[2,2]); # giving size as Array, non-efficient Matlab way
+"""
+onesM(dim::Integer)=ones(dim,dim)
+onesM(a::Array)=ones(Tuple(a))
+onesM(T::Type,a::Array)=ones(T::Type,Tuple(a))
+onesM(args...) = ones(args...)
+################################################################
+"""
+    randM(idx)         # square idx*idx matrix
+    randM(Type,idx)    # square idx*idx matrix
+    randM(sizeAsArray) # non-efficient Matlab way
+    randM(Type,sizeAsArray) # non-efficient Matlab way
+
+
+# Example
+mRand0=randM(2); # same as rand(2,2)
+
+mRand1=randM(2,2); # same as rand(2,2)
+
+mRand2=randM(Int32,2,2); # same as rand(Int32,2,2)
+
+mRand3=randM((2,2)); # giving size as Tuple
+
+mRand4=randM(Int32,(2,2)); # giving size as Tuple and also the Int32 type
+
+mRand5=randM([2,2]); # giving size as Array, non-efficient Matlab way. Array should be Integer otherwise you will get errors.
+
+mRand6=randM(Int32,[2,2]); # giving size as Array, non-efficient Matlab way
+"""
+randM(dim::Integer)=rand(dim,dim)
+randM(a::Array)=rand(Tuple(a))
+randM(T::Type,a::Array)=rand(T::Type,Tuple(a))
+randM(args...) = rand(args...)
+################################################################
+"""
+    eyeM(idx)         # square idx*idx matrix
+    eyeM(Type,idx)    # square idx*idx matrix
```

Comment:
Technically it should be `eyeM(T::Type, idx)` .

In Julia we often write these as `eyeM([T::Type,] idx)` indicating that `T` is an optional input, but since Matlab doesn't have this convention, I think both are accepted as long as we're consistent.

Reply:
Oh those are just my function prototype example for documentation, that's why I didn't write `::Type` .

I wanted to ask this. So if I write `eyeM([T::Type,] idx)` does it cover both `eyeM(T::Type, idx)` and `eyeM(idx)` methods?

Reply:
> So if I write eyeM([T::Type,] idx) does it cover both eyeM(T::Type, idx) and eyeM(idx) methods?

Yes.

Reply:
This gives me an error of 
```julia
eyeM([T::Type,] idx)
syntax: missing comma or ) in argument list
```

Reply:
Ha, it's the convention of the docstring; not a julia grammar.

Reply:
Oh right. As I said, I am writing this package mostly for Matlab users, so I don't want the documentation to be complex. The way I wrote it is much more intuitive.